XMonad/Hooks/DynamicIcons.hs

```diff
@@ -0,0 +1,141 @@
+{-# LANGUAGE TupleSections #-}
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  XMonad.Hooks.DynamicIcons
+-- Copyright   :  (c) Will Pierlot <willp@outlook.com.au>
+-- License     :  BSD3-style (see LICENSE)
+--
+-- Maintainer  :  Will Pierlot <willp@outlook.com.au>
+-- Stability   :  unstable
+-- Portability :  unportable
+--
+-- a set of utilities that allows easy manipulation of Dynamic Log to allow
+-- for dynamic icons
```

Comment:
```suggestion
-- Dynamically change workspace icons based on the contents of the workspace
```

Reply:
This description could really benefit from having an example or something. I still don't really know what a "workspace icon" is or where would I want/expect to have one. :-/

Reply:
I guess that's fair, yea. How'd you imagine an example here though? An image?

Reply:
An image would be nice, but since the examples here use emoji icons instead of xmobar image icons, the example could be entirely textual.

But the main problem is that "Dynamically change workspace icons" assumes "workspace icons" is a thing xmonad users already know about, and that's just not true. The implementation looks like it probably replaces workspace names in the status bar with "icons" based on what windows there are, but it's not at all clear whether it really replaces them, appends them to the workspace name/number, or something. So a better description, ideally accompanied by an example, would really really help here.

(And if the author is not a native English speaker or something, it's perfectly fine to ask for help. I personally have quite a lot of work in progress already so I'm not really keen to play with this now, but if some of you guys doing the review were willing to try this in your setup and help with the docs, it would be awesome. ;-))

Reply:
Unresolved this, as there still seems to be some need for further work. @auscyberman sorry i guess :P

Reply:
> An image would be nice, but since the examples here use emoji icons instead of xmobar image icons, the example could be entirely textual.
> 
> But the main problem is that "Dynamically change workspace icons" assumes "workspace icons" is a thing xmonad users already know about, and that's just not true. The implementation looks like it probably replaces workspace names in the status bar with "icons" based on what windows there are, but it's not at all clear whether it really replaces them, appends them to the workspace name/number, or something. So a better description, ideally accompanied by an example, would really really help here.
> 
> (And if the author is not a native English speaker or something, it's perfectly fine to ask for help. I personally have quite a lot of work in progress already so I'm not really keen to play with this now, but if some of you guys doing the review were willing to try this in your setup and help with the docs, it would be awesome. ;-))

I think i said, i changed it to not changed the icons, but just change the text of it